src/components/ImageUpload.vue

```diff
@@ -14,9 +14,7 @@ import modelHelpers from './../store/modules/models/helpers';
 export default {
   name: 'ImageUpload',
   mounted() {
-    window.eventBus.$on('uploadImage', (event) => {
-      this.$refs.imageInput.click();
-    });
+    window.eventBus.$on('uploadImage', this.promptUpload.bind(this));
```

Comment:
Why do we use bind here?

Reply:
Here, we're using `bind()` instead of writing

```
const that = this;
window.eventBus.$on('uploadImage', () => { that.promptUpload(); });
```

We want to make sure that `promptUpload` has the correct context when we call it from the event handler, even though the event handler will overwrite the `this` context.